app/ui/shared/widgets/search.jsx

```diff
@@ -62,9 +72,20 @@ class Search extends Component {
     this.input.select();
   }
 
-  handleClick = () => {
+  handleClick = e => {
     this.select();
-    this.props.onClick();
+    this.props.onClick(e);
+  }
+
+  handleKeyDown = e => {
+    this.setState({ interaction: 'user-is-typing' });
```

Comment:
I think consts for the interaction types would be good here -- elsewhere, like sagas, can remain primitives